graphistry/dgl_utils.py

```diff
@@ -295,85 +306,137 @@ def _convert_edgeDF_to_DGL(self, node_column: Optional[str] = None, weight_colum
         res._mask_edges()
-        return res  # have to return despite inplace flag
+        return res
 
-    def build_dgl_graph(
+    def build_gnn(
         self,
-        node_column: str = None,
-        weight_column: str = None,
         X_nodes: XSymbolic = None,
         X_edges: XSymbolic = None,
         y_nodes: YSymbolic = None,
         y_edges: YSymbolic = None,
-        use_node_scaler: str = None,
-        use_edge_scaler: str = None,
+        weight_column: str = None,
+        train_split: float = 0.8,
+        use_node_scaler: str = "robust",
+        use_edge_scaler: str = "robust",
+        device: str = "cpu",
         inplace: bool = False,
     ):
+        """
+        Builds GNN model using (DGL)[https://www.dgl.ai/]
+
+        Will auto-featurize, and if no explicit edges are found, automatically UMAP to produce implicit edges.
+        ________________________________________________________________________________________________________________
+
+        :param X_nodes: Which node dataframe columns to featurize. If None, will use all columns.
+                If passing in explicit dataframe, will set them as attributes.
+        :param X_edges: Which edge dataframe columns to featurize. If None, will use all columns.
+                If passing in explicit dataframe, will set them as attributes.
+        :param y_nodes: Optional target column from nodes dataframe.
+        :param y_edges: Optional target column from edges dataframe
+        :param weight_column: Optional Weight column if explicit edges table exists with said weights.
+                Otherwise, weight_column is inhereted by UMAP.
+        :param train_split: Randomly assigns a train and test mask according to the split value, default 80%.
+        :param use_node_scaler: selects which scaling to use on featurized nodes dataframe. Default `robust`
+        :param use_edge_scaler: selects which scaling to use on featurized edges dataframe. Default `robust`
+        :param device: device to run model, default `cpu`, with `gpu` the other choice. Can be handled in outer scope.
+        :param inplace: default, False, whether to return Graphistry instance in place or not.
+
+        """
 
         if inplace:
             res = self
         else:
             res = self.bind()
 
-        res.dgl_lazy_init()
+        res.dgl_lazy_init(train_split=train_split, device=device)
 
-        m = res.materialize_nodes()
+        try:
+            m = res.materialize_nodes()
+        except Exception as e:
+            logger.debug(e)
+            logger.info(f'No edges found, please call g.umap(scale=10, ...) to generate implicit edges')
+            raise
+        
         X_nodes_resolved = resolve_X(m._nodes, X_nodes)
         y_nodes_resolved = resolve_y(m._nodes, y_nodes)
+
+        # here we check if edges are from UMAP, at which point X_edges should be none:
+        if list(res._edges.columns) == ["_src_implicit", "_dst_implicit", "_weight"]:
```

Comment:
Why are we dropping these? Why aren't we forcing the user to make this decision?